### PR TITLE
Added Calendar using react-big-calendar

### LINKED ISSUE #163: Feature Request: Create a Community Calendar section
### Is your feature request related to a problem? Please describe.

As of now, we have an initiatives page where our upcoming meetups and events are documented and organized: https://community.moja.global/initiatives

However these are hard to maintain, and are not UX-friendly. We would like to have a community calendar implementation so that it becomes easy for us to know and understand our upcoming meetups and events for a general accessibility.

### Describe the solution you'd like.

We can potentially use:

- https://dptoot.github.io/react-event-calendar/
- https://www.npmjs.com/package/react-full-event-calendar
- https://www.npmjs.com/package/react-big-calendar

### Describe alternatives you've considered

_No response_

### Additional context.

We would like the data to be supplied by either a JSON file which can be easily edited or a Google Sheet which can be maintained by both technical & non technical users.

### PR DESCRIPTION
**Pull request by Palaksharma23**
Fixes #163

## Description
I have added a Calendar using react-big-calendar

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue).

image
image

### CODE REVIEW

**Comment by Palaksharma23** [comment]
@sohamsshah can you please once review my pull request

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Hi @Palaksharma23 👋 

Exceptional work and effort! 

A few initial comments from my side: 

- Lint the codebase to mitigate linting issues and errors.
- Use a JSON file to have all the events (including past and upcoming events)
- Use something like [Luxon](https://moment.github.io/luxon/#/) to allow folks across different time zones to understand the meeting times with respect to their clock.
- Use UTC timezones throughout for uniformity. 
- Test for responsiveness. 

Tagging, @sohamsshah, our UI guru for a more detailed feedback. Great work 👏

**Comment by Palaksharma23** [comment]
Thank you @HarshCasper sir 😀
I will make commits for json file event management , UTC timezone and linting issues. 
But i will have to read about Luxon :)
I will do it asap

File: website/src/pages/initiatives.js [hunk 1]
```diff
@@ -1,10 +1,45 @@
...
+};
+const localizer = dateFnsLocalizer({
+  format,
+  parse,
+  startOfWeek,
+  getDay,
+  locales,
+});
+// format for adding a meeting is new Date(year , month-1 , date)
+const events = [
```
Comment (sohamsshah) [inline website/src/pages/initiatives.js:26]: The events array must have following fields:
```
{
    title: 'Documentation Meeting',
    start: new Date(2021, 11, 2, 16, 30, 0), // should be having hour and minutes along with the time
    end: new Date(2021, 11, 2, 17, 0, 0),
    googleCalendarLink: 'link-to-google-calnedar', // onClick: this should take the user to the Google Calendar of the Moja Global Meetings
  },
  ```
  The events array should be placed `websiteData/meetings.js` and the data can be safely replaced with new ones. The array can be imported in the `initiatives.js` component.

File: website/src/pages/initiatives.js [hunk 2]
```diff
@@ -71,37 +106,34 @@ const Community = () => {
           </div>
         </section>
         <section id="meetings">
-          <h3 className={styles.communitySubtitle}>Our Meetings</h3>
-          <div className={styles.communityRow}>
-            <div className={styles.communityMeeting}>
+        <h3 className={styles.communitySubtitle}>Our Meetings</h3>
+           <div>
+            <div> 
               <h5 className={styles.meetingTitle}>Past Meetings</h5>
```
Comment (sohamsshah) [inline website/src/pages/initiatives.js:112]: We no longer need the `Past Meetings` section. Only one calendar for the Meetings should be enough to track them.

File: website/src/pages/initiatives.js [hunk 3]
```diff
@@ -71,37 +106,34 @@ const Community = () => {
...
-                height={500}
-                itemCount={pastMeetings.length}
-                itemSize={100}
-                width="90%"
-                itemData={{ isCompleted: true }}
-              >
-                {MeetingCard}
-              </List>
+                
+                  <Calendar className={styles.Calendar}
```
Comment (sohamsshah) [inline website/src/pages/initiatives.js:114]: The Styles currently fail in `Dark Mode` toggle and the I feel that color schemes and contrast can be improved. Small UI changes can be implemented to make it look even better.

File: website/src/pages/initiatives.js [hunk 4]
```diff
@@ -71,37 +106,34 @@ const Community = () => {
...
+                    localizer={localizer}
```
Comment (sohamsshah) [inline website/src/pages/initiatives.js:115]: The Props which `<Calendar />` component shall accept:
```
<Calendar 
  className={styles.Calendar}
  localizer={localizer}
  events={events}
  selectable={true}
  views={['month', 'week', 'agenda']} // max number of meetings per day is 1, so it doesn't make sense to have day view
  onSelectEvent={(event) => redirect(event.googleCalendarLink)} // onSelectEvent performs an action when clicked on the event in the calendar
  style={{ height: 400, width: '100%' }} // can be altered based on the implemented style
/>
  ```

**Review by sohamsshah — CHANGES_REQUESTED** [review]
Hey @Palaksharma23 , this is really appreciable work! 👏
I would suggest you to take time, understand and brain storm upon the review comments and focus on creating/solving with cleaner UI similar to [layer5 calendar](https://layer5.io/community/calendar). We don't want that fancy, but it is better to take time and create better UI so it requires less iterations for the implemented feature.

Once the changes are successfully added, we can safely delete the `PastMeetingCard` component and all related imports in the codebase.

As of now, the meetings section (past and upcoming) consisting of tables is now replaced with `React Big Calendar` with events listed in calendar form, which is definitely a good intuitive UI practice.

Thanks for working on it. Please let me know if you need any help in implementation or doubts for this.

**Comment by Palaksharma23** [comment]
thank you sir , i will do that @sohamsshah

**Comment by Palaksharma23** [comment]
@sohamsshah  sir can you please once review my last commit. 
i have added an add to calendar button which will link to google calendar of moja global community website , i have now provided a dummy link because i didn't have google calendar link and also i have added a double click functionality to the events of calendar which will also lead to google calendar of moja global community website

**Comment by Palaksharma23** [comment]
https://user-images.githubusercontent.com/87171452/146358906-a91bf991-6bc4-4ee3-8ad0-0b5e5a257b9e.mp4

**Comment by Palaksharma23** [comment]
https://user-images.githubusercontent.com/87171452/146359084-7a4b649e-b004-4794-a8a8-b8b1aac5917f.mp4

**Review by sohamsshah — APPROVED** [review]
LGTM @Palaksharma23 ! Thanks for working on this! 🎉

**Comment by Palaksharma23** [comment]
@sohamsshah  thank you sir , sure i will remove the merge conflicts

**Comment by Palaksharma23** [comment]
Done sir @ sohamshah :) 
Thanku so much for helping me with css ☺

**Comment by sohamsshah** [comment]
> Done sir @sohamsshah :) Thanku so much for helping me with css ☺

Thanks for working on this. Appreciable work! 🔥

**Pull request merged**